sdk/storage/azure-storage-file/tests/test_file_async.py

```diff
@@ -306,6 +313,44 @@ def test_create_file_with_metadata_async(self):
         loop = asyncio.get_event_loop()
         loop.run_until_complete(self._test_create_file_with_metadata_async())
 
+    async def test_create_file_when_file_permission_is_too_long(self):
+        file_client = await self._get_file_client()
+        permission = str(self.get_random_bytes(8 * 1024 + 1))
+        with self.assertRaises(ValueError):
+            await file_client.create_file(1024, file_permission=permission)
```

Comment:
I don't think this test is being run? I think you need to add the synchronous function:
```
def test_create_file_when_file_permission_is_too_long:
    loop = asyncio.get_event_loop
    loop.run_until......
```

Reply:
Ah yes you are right, I thought there isn't an request sent out so didn't wrap it with an event loop